crypto/src/utils.rs

```diff
@@ -0,0 +1,214 @@
+        let mut eraser = Eraser::new();
+        eraser.add_suite::<Sodium>();
+        let restored = eraser
+            .restore(&self.secret_key)
+            .map_err(|_| Error::new(ErrorKind::Other, "Couldn't restore a secret key"))?;
+        assert_eq!(restored.len(), SEED_LENGTH);
+        let seed_bytes = restored
+            .open(pass_phrase)
+            .map_err(|_| Error::new(ErrorKind::Other, "Couldn't open an encrypted key"))?;
+        let seed = Seed::from_slice(&seed_bytes[..])
+            .ok_or_else(|| Error::new(ErrorKind::Other, "Couldn't create seed from slice"))?;
+        let (public_key, secret_key) = gen_keypair_from_seed(&seed);
+
+        if self.public_key == public_key {
+            Ok((public_key, secret_key))
+        } else {
+            Err(Error::new(ErrorKind::Other, "Different public keys"))
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    extern crate tempdir;
+
+    use self::tempdir::TempDir;
+    use super::*;
+    use toml;
+
+    #[test]
+    fn test_create_and_read_keys_file() {
+        let dir = TempDir::new("test_utils").expect("Couldn't create TempDir");
+        let file_path = dir.path().join("private_key.toml");
+        let pass_phrase = b"passphrase";
+        let pk1 = generate_keys_file(file_path.as_path(), pass_phrase).unwrap();
+        let (pk2, _) = read_keys_from_file(file_path.as_path(), pass_phrase).unwrap();
+        assert_eq!(pk1, pk2);
+    }
+
+    #[test]
+    fn test_encrypt_decrypt() {
+        let pass_phrase = b"passphrase";
+        let (pk, sk) = gen_keypair();
+        let keys = EncryptedKeys::encrypt(pk, &sk, pass_phrase).expect("Couldn't encrypt keys");
+        let (_, decrypted_sk) = keys.decrypt(pass_phrase).expect("Couldn't decrypt key");
+        assert_eq!(sk, decrypted_sk);
+    }
+
+    #[test]
+    fn test_decrypt_from_file() {
+        let pass_phrase = b"passphrase";
+        let file_content = r#"
+public_key = '2e9d0b7ff996acdda58dd786950dec7361d3d81fd188cb250fd0cab2d064aaf8'
+
+[secret_key]
+ciphertext = '7fbb51090742482da42816b2c908ff61c470a19ca1b984014c7ac37dd46ef1ef'
+mac = '862f27c67b07f9665628b6f9a72a1c20'
+kdf = 'scrypt-nacl'
+cipher = 'xsalsa20-poly1305'
+
+[secret_key.kdfparams]
+salt = '2ee70102a15aff032523a5df91e435172ef003ad9898a3a5eb2f5af447d28b63'
+memlimit = 16777216
+opslimit = 524288
+
+[secret_key.cipherparams]
+iv = '374c8dc0ab8d753ae0515f485e24f6c76b469cde3dee285c'
+        "#;
+
+        let keys: EncryptedKeys =
+            toml::from_str(file_content).expect("Couldn't deserialize content");
+        let (public_key, secret_key) = keys.decrypt(pass_phrase).expect("Couldn't decrypt key");
+        assert_eq!(
+            public_key.to_hex(),
+            "2e9d0b7ff996acdda58dd786950dec7361d3d81fd188cb250fd0cab2d064aaf8"
+        );
+        assert_eq!(secret_key.to_hex(),
+"47782139daefd1c1764d9ed0faa3e8e591c89a9c4e786758d196ed5041ca9e572e9d0b7ff996acdda58dd786950dec7361d3d81fd188cb250fd0cab2d064aaf8")
```

Comment:
Nit: It's possible to break line here, right?

```rust
assert_eq!(
    secret_key.to_hex(),
    "47782139daefd1c1764d9ed0faa3e8e591c89a9c4e786758d196ed5041ca9e57\
     2e9d0b7ff996acdda58dd786950dec7361d3d81fd188cb250fd0cab2d064aaf8"
)
```

Reply:
I suppose yes 😄 But `cargo fmt` hasn't done that.